spark/src/main/java/org/opensearch/sql/spark/dispatcher/DatasourceEmbeddedQueryIdProvider.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.dispatcher;
+
+import org.opensearch.sql.spark.asyncquery.model.AsyncQueryId;
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Generates QueryId by embedding Datasource name and random UUID */
+public class DatasourceEmbeddedQueryIdProvider implements QueryIdProvider {
+
+  @Override
+  public String getQueryId(DispatchQueryRequest dispatchQueryRequest) {
+    return AsyncQueryId.newAsyncQueryId(dispatchQueryRequest.getDatasource()).getId();
```

Comment:
`AsyncQueryId` is only used here? Could you consider rename it to `DatasourceEmbeddedQueryId`? Or is it also used elsewhere?

Reply:
Good point. Will fix.